src/taipy/gui/gui.py

```diff
@@ -1369,8 +1370,12 @@ def __render_page(self, page_name: str) -> t.Any:
         if hasattr(self, "on_navigate") and callable(self.on_navigate):
             try:
                 nav_page = self.on_navigate(self.__get_state(), page_name)
-                if not isinstance(nav_page, str):
-                    warnings.warn(f"on_navigate() returned a invalid page name '{nav_page}'.")
+                if nav_page != page_name:
+                    if isinstance(nav_page, str):
+                        if self._navigate(nav_page):
+                            return ("Root page cannot be re-routed by on_navigate", 302)
```

Comment:
"on_navigate" -> "on_navigate()."

What does this mean? re-routed??
You cannot navigate from '/' to another page?
So that would be "Cannot navigate from root page in on_navigate()."

Reply:
we cannot replace the destination page in on_navigate when the root page is requested

Reply:
I could even not call on_navigate